Sources/Datadog/Logger.swift

```diff
@@ -57,7 +57,7 @@ public enum LogLevel: Int, Codable {
 ///
 public typealias DDLogger = Logger
 
-public class Logger {
+public class Logger: NSObject {
```

Comment:
This is needed to inject the logger in the crash reporting objc plugin protocol..

Reply:
IMO it's definitely not a way to go - this will add the entire [`NSObject` boilerplate](https://developer.apple.com/documentation/objectivec/nsobject) to `Logger` object in user's code. I believe it will not only sink its APIs discoverability in auto-completion but also change the performance of `Logger`.

I'd consider different approaches, e.g.:
* having public `DDCrashReportingLogger` interface that `Logger` will conform to (internally);
* having public `DDCrashReportingLogger` type that will wrap the `Logger`.

Also, mind that this:
```swift
@objc
public protocol DDCrashReportingPluginType: AnyObject
```
is just a convenience in `Datadog` (swift) so we don't have to provide separate interface from `DatadogObjc`. What we struggle with here is just a result of this convenience shortcut but issues like this are commonly solved in `DatadogObjc` by providing `@objc` wrappers for all public Swift APIs. I think it's not worth looking for a solution in `DatadogObjc` as the problem is relatively small 😉. But such small problem should not lead to bypassing the `DatadogObjc` → `Datadog` separation and introducing `NSObject` cost to `Logger`.

Reply:
Indeed, making `Logger` a `NSObject` subclass is not the right way. But The first proposed approach won't work, the `DDCrashReportingLogger` will need to be objc and we will end up with the same issue with `Logger` being pure Swift. We can create a objc `DDCrashReportingLogger` wrapper for `Logger`, but this seems to be quite excessive for just adding some logs.
I'm also wondering if we could just make the `DDCrashReportingPluginType` a pure Swift protocol, I don't see this abstraction being used by an objc integration, it is mostly for our own use. WDYT?

Reply:
> I'm also wondering if we could just make the `DDCrashReportingPluginType` a pure Swift protocol, I don't see this abstraction being used by an objc integration, it is mostly for our own use. WDYT?

Yes, this could work and it sounds good 👍. There is no use case for providing implementation to this protocol in Objc and seems that objc constraint was added to it for no reason.

Reply:
Damn.. it's objc because of [this](https://github.com/DataDog/dd-sdk-ios/blob/master/Sources/DatadogObjc/DatadogConfiguration%2Bobjc.swift#L197).
To follow the current pattern, we will need to create a bridge to `DDCrashReportingPluginType` in `DatadogObjc`. This is a fairly large refactoring with impact on our public interface.. I will rather add logs to the integration layer instead of within the plugin.

Reply:
Indeed 😞. And definitely agree - tracing the plugin is separate challenge and the goal of `RUMM-1796` was to trace the integration logic itself. Maybe we can create separate task for exporting logs from the plugin later?